src/ipns/dns.rs

```diff
@@ -18,7 +18,7 @@ use thiserror::Error;
 #[error("no dnslink entry")]
 pub struct DnsLinkError;
 
-type FutureAnswer = Pin<Box<dyn Future<Output = Result<Answer, io::Error>>>>;
```

Comment:
note: this one gave me a bit of a headache - warp was pointing to a **very** long issue with the HTTP call, but this was actually the root cause of the problem

Reply:
Yeah I wonder if the future should be just written open as `async fn` to be simpler in every way, but haven't tried this myself ... for a while at least. Though, there's nothing obvious causing this, perhaps the path inside domain-resolve is just so long.